crates/core/src/client/mod.rs

```diff
@@ -0,0 +1,324 @@
+
+    /// Set the time out for establishign the initial http connection in milliseconds.
+    ///
+    /// By default the timeout is 30 seconds.
+    pub fn set_dead_render_timeout_ms(&self, timeout: u64) {
+        let mut config = self.config.lock().unwrap();
+        config.dead_render_timeout = timeout;
+    }
+
+    /// Set the time out for awaiting responses from the websocket in milliseconds.
+    ///
+    /// By default the timeout is 5 seconds.
+    pub fn set_websocket_timeout_ms(&self, timeout: u64) {
+        let mut config = self.config.lock().unwrap();
+        config.websocket_timeout = timeout;
+    }
+
+    /// Sets the '_format' arg set upon fetching the dead render and upon
+    /// establishing the websocket connection.
+    ///
+    /// On android this defaults to [Platform::Jetpack], on apple vendored
+    /// devices this defaults to [Platform::Swiftui], everywhere else it
+    /// defaults to "unknown", which will cause a connection failure.
+    pub fn set_format(&self, format: Platform) {
+        let mut config = self.config.lock().unwrap();
+        config.format = format;
+    }
+
+    /// Retrieve the value set in [LiveViewClientBuilder::set_channel_join_params]
+    pub fn channel_join_params(&self) -> Option<HashMap<String, JSON>> {
+        let config = self.config.lock().unwrap();
+        config.join_params.clone()
+    }
+
+    /// Returns the current log level setting.
+    pub fn log_level(&self) -> LogLevel {
+        let config = self.config.lock().unwrap();
+        config.log_level
+    }
+
+    /// Returns the current dead render timeout in milliseconds.
+    pub fn dead_render_timeout(&self) -> u64 {
+        let config = self.config.lock().unwrap();
+        config.dead_render_timeout
+    }
+
+    /// Returns the current websocket timeout in milliseconds.
+    pub fn websocket_timeout(&self) -> u64 {
+        let config = self.config.lock().unwrap();
+        config.websocket_timeout
+    }
+
+    /// Returns the current platform format setting.
+    pub fn format(&self) -> Platform {
+        let config = self.config.lock().unwrap();
+        config.format.clone()
+    }
+
+    /// Attempt to establish a new, connected [LiveViewClient] with the param set above
+    pub async fn connect(&self, url: String) -> Result<LiveViewClient, LiveSocketError> {
+        let config = self.config.lock().unwrap().clone();
+        let inner = LiveViewClientInner::initial_connect(config, url).await?;
+
+        Ok(LiveViewClient { inner })
+    }
+}
+
+#[derive(uniffi::Object)]
+pub struct LiveViewClient {
+    inner: LiveViewClientInner,
+}
+
+#[uniffi::export(async_runtime = "tokio")]
+impl LiveViewClient {
+    pub async fn reconnect(&self, url: String) -> Result<(), LiveSocketError> {
+        self.inner.reconnect(url, Default::default()).await?;
+        Ok(())
+    }
+
+    pub async fn upload_files(&self, files: Vec<Arc<LiveFile>>) -> Result<(), LiveSocketError> {
+        let chan = self.inner.channel()?;
```

Comment:
This is a stop gap, I don't want to fully remove the `LiveChannel` just yet, but this should be a function on the `LiveClientChannel` in a future update. That will allow all replies to be forwarded to the event loop.